service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java

```diff
@@ -322,4 +294,146 @@ protected final CorsHandler createCorsHandler(final String allowedOrigin, final
                 .allowedHeader(HttpHeaders.IF_MATCH.toString())
                 .exposedHeader(HttpHeaders.ETAG.toString());
     }
+
+    /**
+     * Creates a new <em>OpenTracing</em> span for tracing the execution of a service operation.
+     *
+     * @param operationName The operation name that the span should be created for.
+     * @param spanContext Existing span context.
+     * @param tracer the Tracer instance.
+     * @param className The class name to insert in the Span.
+     * @return The new {@code Span}.
+     * @throws NullPointerException if operationName is {@code null}.
+     */
+    public static final Span newChildSpan(final String operationName, final SpanContext spanContext,
```

Comment:
I would leave span related methods in the separate class, as they are used outside of these endpoints.